subsys/bluetooth/mesh/net.c

```diff
@@ -882,8 +882,12 @@ static void ivu_refresh(struct k_work *work)
 			store_iv(true);
 		}
 
-		k_work_reschedule(&bt_mesh.ivu_timer, BT_MESH_IVU_TIMEOUT);
-		return;
```

Comment:
`ivu_timer` should be running all the time, otherwise it will never exceed 96 hours.

Reply:
No need to check time if `ivu_duration` is longer than `BT_MESH_IVU_TIMEOUT`.
It will be started again here: https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/mesh/net.c#L326
if the network layer meets IV update procedure condition.

Reply:
The problem is that it is still possible to lose IV update.

Reply:
After the first recovery, if we lose the next IV update, we need to be able to carry out the next recovery in 192 hours

Reply:
Yep, in this case recovery logic should be triggered according to spec
https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/mesh/net.c#L275

Reply:
`ivi_was_recovered` should be clear before next recovery.

Reply:
This part is what I agree and I approved of that. But adding checking within the scheduled work with permanent restarting IVU refresh time just to poll 192 hours expiration is some workaround. There is one place where mesh stack makes the decision to start or to not start IV update procedure. This is `bt_mesh_net_iv_update`. Adding implicit and parallel clean of `ivi_was_recovered` flag is a bit messy.  
You can check `bt_mesh.ivu_duration` in place of making decision about triggering the recovery procedure: https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/mesh/net.c#L277